sklearn/utils/testing.py

```diff
@@ -882,3 +903,147 @@ def check_docstring_parameters(func, doc=None, ignore=None, class_name=None):
+
+                assert u_dict['type_definition'] == type_definition, msg1
+                assert u_dict['description'] == description, msg2
+            else:
+                add_dict = {'type_definition': type_definition,
+                            'description': description}
+                type_dict[name] = add_dict
+
+    return type_dict
+
+
+def assert_consistent_docs(objects,
+                           include_params=True, exclude_params=None,
+                           include_attribs=True, exclude_attribs=None,
+                           include_returns=True, exclude_returns=None):
+    """
+    Checks consistency between the docstring of ``objects``.
+
+    Checks if types and descriptions of Parameters/Attributes/Returns are
+    identical across ``objects``. Raises AssertionError if found otherwise.
+
+    Parameters
+    ----------
+    objects : list
+        The list of objects that may be either ``NumpyDocString`` instances or
+        objects (classes, functions, descriptors) with docstrings that can be
+        parsed as numpydoc.
+
+    include_params : list, False or True (default)
+        List of Parameters to be included. True, for including all parameters.
+
+    exclude_params : list or None (default)
+        List of Parameters to be excluded. Set only if include_params is True.
+
+    include_attribs : list, False or True (default)
+        List of Attributes to be included. True, for including all attributes.
+
+    exclude_attribs : list or None (default)
+        List of Attributes to be excluded. Set only if include_attribs is True.
+
+    include_returns : list, False or True (default)
+        List of Returns to be included. True, for including all returns.
+
+    exclude_returns : list or None (default)
+        List of Returns to be excluded. Set only if include_returns is True.
+
+    Notes
+    -----
+    This function asserts that any Parameters/Returns/Attributes entries having
+    the same name among ``objects`` docstrings also have the same type
+    specification and description.
+    It compares only those entries having the same name in the docstring, does
+    no comparison if an entry in a docstring is unique.
+
+    Examples
+    --------
+    >>> from sklearn.metrics import (mean_absolute_error, mean_squared_error,
+    ... median_absolute_error)
+    >>> from sklearn.utils.testing import assert_consistent_docs
+    ... # doctest: +SKIP
+    >>> assert_consistent_docs([mean_absolute_error, mean_squared_error],
+    ... include_params=['y_true', 'y_pred', 'sample_weight'],
+    ... exclude_attribs='*', exclude_returns='*')  # doctest: +SKIP
+    >>> assert_consistent_docs([median_absolute_error, mean_squared_error],
+    ... include_params='*', exclude_attribs='*', exclude_returns='*')
+    ... # doctest: +SKIP
+    Traceback (most recent call last):
+        ...
+    AssertionError: Parameter y_true of mean_squared_error has inconsistency.
+
+    """
+    from numpydoc import docscrape
+
```

Comment:
Should validate include and exclude make sense

Reply:
It would be helpful and do no harm, I think we should add it.